Package can't be built for Debian Jessie

After being able to successfully compile osquery 1.6.3, make packages fails at:

CMakeFiles/packages.dir/build.make:50: recipe for target 'CMakeFiles/packages' failed
make[4]: *** [CMakeFiles/packages] Error 1
CMakeFiles/Makefile2:189: recipe for target 'CMakeFiles/packages.dir/all' failed
make[3]: *** [CMakeFiles/packages.dir/all] Error 2
CMakeFiles/Makefile2:197: recipe for target 'CMakeFiles/packages.dir/rule' failed
make[2]: *** [CMakeFiles/packages.dir/rule] Error 2
Makefile:213: recipe for target 'packages' failed
make[1]: *** [packages] Error 2
Makefile:115: recipe for target 'packages' failed
make: *** [packages] Error 2

$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support/"
BUG_REPORT_URL="https://bugs.debian.org/"

--- a/CMake/Packages.cmake
+++ b/CMake/Packages.cmake
@@ -11,14 +11,29 @@ if(APPLE)
   )
 elseif(LINUX)
   if(DEBIAN_BASED)
+    # Set basic catch-alls for debian-based systems.
     set(PACKAGE_TYPE "deb")
-    set(PACKAGE_ITERATION "1.ubuntu")
+    set(PACKAGE_ITERATION "1.debian")
     set(PACKAGE_DEPENDENCIES
       "zlib1g"
       "libbz2-1.0"
       "libreadline6"
-      "libgcrypt11"
     )
+
+    # Improve catch-alls for debian or ubuntu.
+    if(OSQUERY_BUILD_PLATFORM STREQUAL "ubuntu")
+      set(PACKAGE_DEPENDENCIES
+        "${PACKAGE_DEPENDENCIES}"
+        "libgcrypt11"
+      )
+    elseif(OSQUERY_BUILD_PLATFORM STREQUAL "debian")
+      set(PACKAGE_DEPENDENCIES
+        "${PACKAGE_DEPENDENCIES}"
+        "libgcrypt20"
+      )
+    endif()
+
+    # Improve package and iterations for each specific distribution.
     if(NOT OSQUERY_BUILD_DISTRO STREQUAL "lucid")
       set(PACKAGE_ITERATION "1.ubuntu10")
       set(PACKAGE_DEPENDENCIES
@@ -27,6 +42,7 @@ elseif(LINUX)
         "libapt-pkg4.12"
       )
     endif()
+
     if(OSQUERY_BUILD_DISTRO STREQUAL "precise")
       set(PACKAGE_ITERATION "1.ubuntu12")
       set(PACKAGE_DEPENDENCIES
@@ -34,8 +50,22 @@ elseif(LINUX)
         "libstdc++6"
         "libudev0"
       )
+    elseif(OSQUERY_BUILD_DISTRO STREQUAL "wheezy")
+      set(PACKAGE_ITERATION "1.debian7")
+      set(PACKAGE_DEPENDENCIES
+        "${PACKAGE_DEPENDENCIES}"
+        "libstdc++6"
+        "libudev0"
+      )
     elseif(OSQUERY_BUILD_DISTRO STREQUAL "trusty")
       set(PACKAGE_ITERATION "1.ubuntu14")
+      set(PACKAGE_DEPENDENCIES
+        "${PACKAGE_DEPENDENCIES}"
+        "libstdc++6 (>= 4.8)"
+        "libudev1"
+      )
+    elseif(OSQUERY_BUILD_DISTRO STREQUAL "jessie")
+      set(PACKAGE_ITERATION "1.debian8")
       set(PACKAGE_DEPENDENCIES
         "${PACKAGE_DEPENDENCIES}"
         "libstdc++6 (>= 4.8)"

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -249,6 +249,10 @@ add_definitions(
 
 if(APPLE)
   LOG_PLATFORM("OS X")
+elseif(OSQUERY_BUILD_PLATFORM STREQUAL "debian")
+  set(DEBIAN_BASED TRUE)
+  set(DEBIAN TRUE)
+  LOG_PLATFORM("Debian")
 elseif(OSQUERY_BUILD_PLATFORM STREQUAL "ubuntu")
   set(DEBIAN_BASED TRUE)
   set(UBUNTU TRUE)

--- a/osquery/database/db_handle.h
+++ b/osquery/database/db_handle.h
@@ -32,7 +32,9 @@ typedef std::shared_ptr<DBHandle> DBHandleRef;
 
 class GlogRocksDBLogger : public rocksdb::Logger {
  public:
-  virtual void Logv(const char* format, va_list ap);
+  // We intend to override a virtual method that is overloaded.
+  using rocksdb::Logger::Logv;
+  void Logv(const char* format, va_list ap) override;
 };
 
 /**

--- a/tools/provision/debian.sh
+++ b/tools/provision/debian.sh
@@ -68,8 +68,12 @@ function main_debian() {
     package libboost-all-dev
     package automake
   fi 
-  
+
   install_google_benchmark
+
+  package rubygems
+  gem_install fpm
+
   install_thrift
   install_rocksdb 
   install_yara

